### PR TITLE
Extend IApplication to provide more generic functionality

### PR DESCRIPTION
**Pull request by sebaszm**

### CODE REVIEW

File: interfaces/INetflix.h [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (artur-gebicz-ml) [inline interfaces/INetflix.h:1]: Shouldn't those changes be in the IApplication.h instead of INetflix.h?

Comment (artur-gebicz-ml) [reply interfaces/INetflix.h:1]: Please do not mix tabs with space. Use consistent indentation (spaces)

Comment (artur-gebicz-ml) [reply interfaces/INetflix.h:1]: Are we going to leave any methods inside the INetflix interface?

**Review by artur-gebicz-ml — CHANGES_REQUESTED** [review]

File: interfaces/INetflix.h (hunk 1 above)
Comment (sebaszm) [reply interfaces/INetflix.h]: too fast, too furious, sorry

**Review by artur-gebicz-ml — APPROVED** [review]

File: interfaces/IApplication.h [hunk 2]
```diff
@@ -23,17 +24,79 @@
...
+            VOICE_SEARCH_RESULT,
+            VISUAL_GESTURE,
+            TOUCH_GESTURE,
+            EPG_GRID,
+            CHANNEL_NUMBER,
+            CHANNEL_ZAP,
+            CHANNEL_BAR,
+            WEB_BROWSER,
+            POWER_ON,
+            POWER_ON_FROM_DEDICATED_BUTTON
```
Comment (HaseenaSainul) [inline interfaces/IApplication.h:60]: ',' is missing here

File: interfaces/IApplication.h [hunk 3]
```diff
@@ -23,17 +24,79 @@
...
+        };
+
+        // @brief Resets application data
+        // @param type Type of reset to perform
+        virtual uint32_t Reset(const resettype type) = 0;
+
+        // @property
+        // @brief Application-specific identification value
+        // @param id ID string (e.g. ABCDEFG=00000005)
+        virtual uint32_t ID(string& id /* @out */) const = 0;
```
Comment (HaseenaSainul) [inline interfaces/IApplication.h:78]: Showing error, since both enum and property use same name 'ID"

Comment (sebaszm) [reply interfaces/IApplication.h]: Renamed to Identifier

File: interfaces/IApplication.h [hunk 4]
```diff
@@ -23,17 +24,79 @@
...
+            APPLICATION_PROMOTION
+        };
+
+        // @brief Resets application data
+        // @param type Type of reset to perform
+        virtual uint32_t Reset(const resettype type) = 0;
+
+        // @property
+        // @brief Application-specific identification string
+        // @param id Identifier string (e.g. ABCDEFG=00000005)
```
Comment (pwielders) [inline interfaces/IApplication.h:77]: I think the equal sign is something we would not like to advertise cause now I think you suggest that netflix should expose ESN=MOS5355500000001 and I think we should only expose MOS53555000000001. The one who uses this interface already has a callsign from where he got this interface e.g. Netflix plugin so my suggestion would be to not expose the detail of the implementation that it is an ESN, but just the ID from the Netflix plugin.

**Review by pwielders — COMMENTED** [review]
Minor remark on a comment in the file..

File: interfaces/IApplication.h (hunk 4 above)
Comment (sebaszm) [reply interfaces/IApplication.h:77]: Perhaps having Netflix ESN as example is unfortunate as it literally is "string equals number".

Comment (sebaszm) [reply interfaces/IApplication.h]: Removed example

**Review by pwielders — APPROVED** [review]

**Review by HaseenaSainul — APPROVED** [review]

**Pull request merged**